### ISSUE TITLE
[New Model]: Support Zyphra/Zamba2-7B

### ISSUE THREAD

**Issue by mgoin** [vllm-project/vllm#9382]
### The model to consider.

Announcement blog: https://www.zyphra.com/post/zamba2-7b

Base model: https://huggingface.co/Zyphra/Zamba2-7B
Instruct tuned: https://huggingface.co/Zyphra/Zamba2-7B-Instruct

![image](https://github.com/user-attachments/assets/bba7f100-f7cf-4284-b8b0-90ed99d9a522)

### The closest model vllm already supports.

Jamba, as it is a mixture of state-space and transformers blocks

> Zamba2-7B-Instruct is a hybrid model composed of state-space ([Mamba2](https://github.com/state-spaces/mamba)) and transformer blocks.

### What's your difficulty of supporting the model you want?

Should be easy once Mamba2 support lands in https://github.com/vllm-project/vllm/pull/9292, however this `use_shared_attention_lora` case seems possibly complex

All of the HF-compatible modeling code can be found here: https://github.com/Zyphra/transformers_zamba2/tree/main/src/transformers/models/zamba2

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by scwall** [comment]
+1

**Comment by yury-tokpanov** [comment]
Hey. Yury here from Zyphra. We have an internal version that works, going to open a PR sometime soon.

**Comment by yury-tokpanov** [comment]
This is done now. Merged into 0.8.x version of vLLM
